sampleSpecs/idref.smithy

```diff
@@ -0,0 +1,12 @@
+$version: "2"
+
+namespace smithy4s.example
+
+structure TestIdRef {
+  @idRef
```

Comment:
suggestion: this relates to another comment, I'd like to see tests for non-struct cases with idrefs. E.g. a custom `string` type, a member of a union / collection etc.